### ISSUE TITLE
CodeCarbon (1) prevents compute node from stopping if error is thrown, (2) runs even if not activated

### ISSUE THREAD

**Issue by MoritzLaurer** [mlco2/codecarbon#305]
* CodeCarbon version: 2.0
* Python version: 3.9
* Operating System: Linux

### Description

I've integrated codecarbon in my script of pytorch tools to track the footprint of training a transformer model with Hugging Face. The tracking sort of works (although this other issue persists: https://github.com/mlco2/codecarbon/issues/298).
I first ran the code on google colab and I'm now running it on a GPU compute cluster from my university. 

I'm now encountering two new issues: 
(1) Codecarbon prevents my compute node from stopping, even when my code produces an error (unrelated to codecarbon). The expected behaviour would be: I submit my script to the compute node and when there is some error, the compute node stops. Since I've installed codecarbon, the compute nodes don't stop running anymore, even if there was an error. The error is just thrown and code carbon then still continues to print emission estimates every 15 sec, although the code should just stop. This is a big issue when using a compute cluster, where compute is expensive and I need it to stop when there is an error. 
(2) codecarbon even seems to run when I'm not starting it in my session. I've installed it for one run. Then I stop that run and reopen the compute node a few hours later. Then I rerun my script, but **without** importing codecarbon or starting any tracker. However, codecarbon is then still outputting tracking info, although I don't want to to run and didn't even activate it. 

=> I have the impression that codecarbon is somehow nesting itself quite deeply in the hardware, so that it's still activated even if I restart the hardware. I don't think that's only an issue with my university compute cluster, because I had the same issue (2) when running and restarting google colabs. 

=> my current solution is to manually uninstall codecarbon because these issues make it unusable for me

### What I Did

```
from codecarbon import OfflineEmissionsTracker
tracker = OfflineEmissionsTracker(country_iso_code="NLD",  log_level='warning', measure_power_secs=300,  # output_dir=TRAINING_DIRECTORY
                                    project_name=f"{DATASET_NAME}-{MODEL_NAME.split('/')[-1]}")
tracker.start()

# .... compute intensive code 

tracker.stop() 


```

**Comment by SaboniAmine** [comment]
Hello! By any chance, would you have activated the [monitor](https://github.com/mlco2/codecarbon#code-carbon-api-alpha) mode of code carbon ?
If you can reproduce, could you check if there is code carbon daemon running on the node (when your program is stopped) ?

**Comment by benoit-cty** [comment]
Hello,
Code Carbon work with a scheduler : https://github.com/mlco2/codecarbon/blob/2a25fa927d32975c549596ec9f1ce3f0d6ed9999/codecarbon/external/scheduler.py#L4
Can you try with stop in a `finally` :
```py
from codecarbon import OfflineEmissionsTracker
tracker = OfflineEmissionsTracker(country_iso_code="NLD",  log_level='warning', measure_power_secs=300,  # output_dir=TRAINING_DIRECTORY
                                    project_name=f"{DATASET_NAME}-{MODEL_NAME.split('/')[-1]}")
tracker.start()

try:
    # .... compute intensive code 
finally:
    tracker.stop() # better in a try/finally block to make sure the program ends!
```

**Comment by qmarcou** [comment]
Hi,
I'm experiencing the same issue using the tracker decorator function around my model fit function in tensorflow :
'''python
@track_emissions(offline=True, country_iso_code="FRA")
fit():
   return
'''
Whenever my fit function fails (e.g in my case it was due to a GPU memory error due to a too large batch size) the job launched through a SLURM scheduler will keep on running until the time limit is reached. I could pinpoint this using an interactive session, and noticed that even when an error occurs, the python script will keep on outputting code-carbon stuff.

It seems that the online version of the decorator implements a try/finally while the offline one does not... Is there any reason for this?

best regards,

**Comment by MoritzLaurer** [comment]
thanks @benoit-cty for your response. It would be great if there were an easier way of fixing this. It seems unnecessarily complicated if everyone running codecarbon on some GPU cluster (like @qmarcou and many others I'm working with a SLURM system) needs to wrap their entire code into an inconvenient try-finally statement, which could cause other downstream issues (haven't tried yet if it actually fixes the issue).

**Comment by benoit-cty** [comment]
Thanks a lot @qmarcou, I've merge in a branch `fix-running-exception` to bump the version and prepare a new release of the package.

**Comment by benoit-cty** [comment]
Closed by the release of V2.1.3 https://pypi.org/project/codecarbon/2.1.3/
